src/mixins/filing-template-mixin.ts

```diff
@@ -571,9 +572,9 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     this.setFilingId(+draftFiling.header.filingId)
 
     // restore Business data
-    this.setEntityType(draftFiling.business.legalType)
-    this.setLegalName(draftFiling.business.legalName)
-    this.setFoundingDate(draftFiling.business.foundingDate)
+    this.setEntityType(draftFiling.business.legalType || this.getBusinessLegalType)
+    this.setLegalName(draftFiling.business.legalName || this.getBusinessLegalName)
+    this.setFoundingDate(draftFiling.business.foundingDate || this.getBusinessFoundingDate)
```

Comment:
Because the draft restoration (or dissolution, below) didn't have legal name, it was being cleared out here. (It was actually loaded via a business fetch in App.vue, so this code here is probably not needed, but that cleanup is out of scope for this quick fix.)